cumulusci/robotframework/pageobjects/PageObject.py

```diff
@@ -0,0 +1,125 @@
+import re
+import os.path
+import sys
+import logging
+from robot.libraries.BuiltIn import BuiltIn
+
+
+class PageObject(object):
+    ROBOT_LIBRARY_SCOPE = "GLOBAL"
+
+    # These must be defined by subclasses
+    PAGE_TYPE = None
+    OBJECT_API_NAME = None
+
+    """Base class to use for all page objects
+
+    Derived classes should define PAGE_TYPE based on the appropriate
+    pageReference type but with the "standard_" prefix removed (eg:
+    objectPage rather than standard_objectPage)
```

Comment:
What's a pageReference type?

Reply:
That's me pretending I know what I'm talking about :-)

The idea is that we have a base class for each lightning component type - see [Lightning Aura Components Developer Guide](https://developer.salesforce.com/docs/atlas.en-us.lightning.meta/lightning/components_navigation_page_definitions.htm). 

To this point I've focused on the "Object Page Type" (hence the file ObjectPages.py elsewhere in this branch). The idea is to have a base definition for object lists, record views, etc. so that when someone is automating a page, they start by inheriting one of our base classes that most closely resembles the page they are automating.

This is definitely something I need more advice on -- is this a good idea, or am I headed in the wrong direction? I think I stumbled on this page when trying to understand the difference between urls like /lightning/r/whatever and /lightning/o/whatever. When I saw a neat little set of documented page types, it seemed like a good way to organize base classes.

While you're free to call it a genius idea :-), I'm perfect open to being told it's over engineering and we really don't need a ton of base classes based on the page type.

Reply:
I'm not sure how it will play out but am willing to go with it for now.